### PR TITLE
Added split_string_list with string as separator.

### PR DESCRIPTION
**Pull request by luca-heltai**
Minor fix, that will allow some cleanup in `Patterns` later on.

### CODE REVIEW

File: tests/base/utilities_01.cc [hunk 1]
```diff
@@ -65,6 +65,18 @@ void test ()
...
+  {
+    const char *p = "alpha;; beta;; gamma ";
+    AssertThrow (Utilities::split_string_list (p, ";;").size() == 3,
+                 ExcInternalError());
+    AssertThrow (Utilities::split_string_list (p, ";;")[0] == "alpha",
+                 ExcInternalError());
+    AssertThrow (Utilities::split_string_list (p, ";;")[1] == "beta",
+                 ExcInternalError());
+    AssertThrow (Utilities::split_string_list (p, ";;")[2] == "gamma",
+                 ExcInternalError());
```
Comment (tamiko) [inline tests/base/utilities_01.cc]: Reminds me that we might consider cleaning up all the places that unnecessarily use `AssertThrow` and `Assert` in the testsuite at some point. A simple `if (...) deallog << ...;` is much more reliable.

Comment (luca-heltai) [inline tests/base/utilities_01.cc]: Do you want me to change this and replace AssertThrows with ifs? In the test suite I thought this was what was basically done in tests.h... 

:)

Comment (tamiko) [inline tests/base/utilities_01.cc]: Please not - leave it as is :-)

Comment (bangerth) [inline tests/base/utilities_01.cc]: I'm a big fan of using Assert also in tests. It makes it so much easier to debug tests than if only the output is wrong. It also avoids the need to store (possibly large) output. In other words, if you know what the exact answer is, I think Assert is the way to go.

File: source/base/utilities.cc [hunk 2]
```diff
@@ -319,6 +318,15 @@ namespace Utilities
   }
 
 
+  std::vector<std::string>
+  split_string_list (const std::string &s,
+                     const char delimiter)
+  {
+    std::string d = ",";
+    d[0] = delimiter;
```
Comment (bangerth) [inline source/base/utilities.cc]: Not wrong, but couldn't you have just written this as `std::string((char[]){delimiter, '\0'})`?

**Pull request merged**